api/core/model_runtime/model_providers/openai_api_compatible/llm/llm.py

```diff
@@ -263,7 +263,7 @@ def get_customizable_model_schema(self, model: str, credentials: dict) -> AIMode
     def _generate(self, model: str, credentials: dict, prompt_messages: list[PromptMessage], model_parameters: dict,
                   tools: Optional[list[PromptMessageTool]] = None, stop: Optional[list[str]] = None,
                   stream: bool = True, \
-                  user: Optional[str] = None) -> Union[LLMResult, Generator]:
+                  user: Optional[str] = None, extra_headers: Optional[dict] = None) -> Union[LLMResult, Generator]:
```

Comment:
This method seems to have an override in Nvidia, OpenRouter, and TogetherAI. Suggest checking if it has any impact.

Reply:
I have made a fix, putting "extra_headers" in credentials, to avoid overriding "_generate" in OAIAPICompatLargeLanguageModel.